core/src/main/java/org/bitcoinj/core/Block.java

```diff
@@ -445,13 +446,20 @@ public String toString() {
      */
     @VisibleForTesting
     public void solve() {
+        Duration fairlyLong = Duration.ofSeconds(5);
+        Stopwatch watch = Stopwatch.start();
         while (true) {
             try {
                 // Is our proof of work valid yet?
                 if (checkProofOfWork(false))
                     return;
                 // No, so increment the nonce and try again.
                 setNonce(getNonce() + 1);
+
+                if (watch.isRunning() && watch.elapsed().compareTo(fairlyLong) > 0) {
```

Comment:
Can the `.isRunning()` check be removed?

Reply:
No, because then it would print thousands of messages until the block is solved.